### PR TITLE
doc: Updates template issue with clearer guidelines.

### PR DESCRIPTION
**Pull request by marcosuma**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): [Engineering Proposal](https://docs.google.com/document/d/10kbAjd3Akf_2j3ymmjZ0yTEWo3IstJrIGl8rAZKD5Zk/edit)

https://jira.mongodb.org/browse/CLOUDP-202434

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [X] Documentation fix/enhancement

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [X] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/Bug_Report.md [hunk 1]
```diff
@@ -6,14 +6,29 @@ about: Something unexpected happen? Report an error, crash, or an incorrect beha
...
+In order to follow the **"one-click reproducible issues" principle**, please follow these guidelines:
+* We should be able to run `terraform plan` and `terraform apply` **with no errors**.
+  * Configurations that return errors when executed will be de-prioritised in favour of the ones that succeed.
+* Share your configuration by **using variables**. Create a `variables.tf` file and share its content.
+* Before opening an issue, you have to try to specifically isolate it to **Terraform MongoDB Atlas** provider by **removing as many dependencies** as possible. If the issue only happens with other dependencies, then:
+  * If other terraform providers are required, please make sure you also include those. Same "one-click reproducible issue" principle applies.
+  * If external components are required to replicate it, please make sure you also provides instructions on those parts.
+
+* If the issue requires **changes from the UI**, please document them preferably with screenshots
+* If the issue involves **version changes** (e.g. `run plan with version 1.k.x then re-run with version 1.k.y`), please make sure you report the detailed sequence on commands to execute.
```
Comment (andreaangiolillo) [inline .github/ISSUE_TEMPLATE/Bug_Report.md]: this list is missing the following info:
              - Terraform Atlas provider version used to reproduce the issue
              - Terraform version used to reproduce the issue
              - Confirmation if Terraform OSS, Terraform Cloud, or Terraform Enterprise deployment

File: .github/ISSUE_TEMPLATE/Bug_Report.md [hunk 2]
```diff
@@ -28,7 +43,7 @@ If you are not running the latest version of Terraform and the MongoDB Atlas Pro
 
 ### Terraform Configuration File 
 <!--
-Paste the relevant parts of your Terraform MongoDB Atlas Provider configuration file info between the ``` marks below or consider creating a GitHub Gist with the information and share it here.
+Paste here all the required parts of your Terraform MongoDB Atlas Provider configuration file below here. You can alternatively consider creating a [**GitHub Gist**](https://gist.github.com/) with the information and share it here.
```
Comment (andreaangiolillo) [inline .github/ISSUE_TEMPLATE/Bug_Report.md]:
```suggestion
Paste here all the required parts of your Terraform MongoDB Atlas Provider configuration file below. You can alternatively consider creating a [**GitHub Gist**](https://gist.github.com/) with the information and share it here.
```

File: .github/ISSUE_TEMPLATE/Bug_Report.md (hunk 1 above)
Comment (marcosuma) [reply .github/ISSUE_TEMPLATE/Bug_Report.md]: Regarding:
```
- Terraform Atlas provider version used to reproduce the issue
- Terraform version used to reproduce the issue
```
maybe we can clarify that the `version.tf` file is also needed (i.e. the `provider` specification). 
I think I specified somewhere else that the *whole* configuration is needed in order for us to be _one-click-away_ from reproducing the issue; but I'll clarify that.

```
- Confirmation if Terraform OSS, Terraform Cloud, or Terraform Enterprise deployment
```
Ok sure, this one I missed it.

Comment (marcosuma) [reply .github/ISSUE_TEMPLATE/Bug_Report.md]: Changes updated. Let me know if that's ok with you.

File: .github/ISSUE_TEMPLATE/Bug_Report.md [hunk 3]
```diff
@@ -6,14 +6,34 @@ about: Something unexpected happen? Report an error, crash, or an incorrect beha
...
+
+**Please note:** In order for us to provide the best experience in supporting our customers, we kindly ask to make sure that all the following sections are correctly filled with all the required information. Our support will **prioritise** issues that contain **all the required** information by using the **"one-click reproducible issues" principle**.
+
+**Please also note:** We try to keep the Terraform MongoDB Atlas Provider issue tracker **reserved for bug reports**. Please ensure you **check open and closed issues first** to ensure your issue hasn't already been reported (if it has been reported add a reaction, i.e. +1, to the issue).
+
+### Guidelines
+
+In order to follow the **"one-click reproducible issues" principle**, please follow these guidelines:
+
+* We should be able to make no changes to your provided script and **be able to run `terraform plan` and `terraform apply` with no errors**.
```
Comment (AgustinBettati) [inline .github/ISSUE_TEMPLATE/Bug_Report.md]:
```suggestion
* We should be able to make no changes to your provided script and **be able to run a local execution reproducing the issue**.
```
Must be able to run with no errors was a little confusing to me as multiple bugs are related to non empty plans or errors in apply

File: .github/ISSUE_TEMPLATE/Bug_Report.md [hunk 4]
```diff
@@ -6,14 +6,34 @@ about: Something unexpected happen? Report an error, crash, or an incorrect beha
...
+  * This means that you should kindly provide us the whole script contatining all the required instructions. This also includes but not limited to:
+    * Terraform Atlas provider version used to reproduce the issue
+    * Terraform version used to reproduce the issue
+  * Configurations that return errors when executed will be de-prioritised in favour of the ones that succeed.
```
Comment (AgustinBettati) [inline .github/ISSUE_TEMPLATE/Bug_Report.md]:
```suggestion
  * Configurations that cannot be properly executed will be de-prioritised in favour of the ones that succeed.
```
same here, suggesting possible rephrase

**Review by AgustinBettati — APPROVED** [review]
LGTM, left a small comment

**Review by andreaangiolillo — APPROVED** [review]
LGTM.

General feedback (no changes are needed, I'm happy to see if the current template works before making any changes): I think the template is too verbose as a user may not want to read all of this info before opening an issue. In the future, we might consider a more concise template that lists the top 3-4 pieces of information users must provide when opening an issue.

**Comment by marcosuma** [comment]
@AgustinBettati updated with your corrections. Thank you!

@andreaangiolillo I agree with your comment, I think too it's verbose but I am not sure if in this specific case it's a problem. 
My rationale is that I consider these instructions like the "Terms & Conditions" that you have when "signing contracts". It's not exactly the same of course but the similarity is that in both cases we acknowledge it's a long text we have to read because we must make sure we write all the necessary information in order to be able to "push back" whenever needed. On the contrary, if we don't write things we'll never be able to ask for something.

This is my rationale. I don't expect actually all users will read it all, but when they don't comply to our guidelines from now on we'll be able to claim that the issue is not ready to be worked. At some point the most frequent issue creators will know our new guidelines.


FYI @andreaangiolillo  @AgustinBettati  I made few other changes or restructuring it.

**Review by andreaangiolillo — APPROVED** [review]

**Pull request merged**